src/components/pages/doc.js

```diff
@@ -128,11 +150,14 @@ function Doc({ initialState, PageTemplate }) {
     return accu;
   }, {});
 
-  const currentRevisionIndex = state.revisions.indexOf(state.currentRevision);
+  const currentRevisionIndex = state.revisions.indexOf(state.currentDocOrRevision);
   const isCurrentRevisionLatestRevision = currentRevisionIndex === state.revisions.length - 1;
 
-  const isExternalDocument = state.currentRevision.origin.startsWith(DOCUMENT_ORIGIN.external);
-  const isEditingDisabled = state.currentRevision.archived || isExternalDocument;
+  const isExternalDocument = state.currentDocOrRevision.origin.startsWith(DOCUMENT_ORIGIN.external);
+  const isEditingDisabled = state.currentDocOrRevision.archived || isExternalDocument || state.type !== DOCUMENT_TYPE.document;
+  const shouldShowViewRevisionsButton = state.type === DOCUMENT_TYPE.document;
+  const shouldShowViewDocumentButton = state.type === DOCUMENT_TYPE.revision;
+  const shouldOfferHardDelete = state.type === DOCUMENT_TYPE.revision;
```

Comment:
Small thing: I'd vote to keep this naming streamlined, between `is` and `should` within this file, in these variables that are defined in a row and act as toggles.

Reply:
I'm trying to fix this one but I'm finding it hard to come up with names that are good with "is" for my situation, care to join? :))

Reply:
How about 
`isViewRevisionsButton[Visible/Shown]`
`isViewDocumentButton[Visible/Shown]`
`isHardDeleteEnabled`
?